src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -947,7 +945,13 @@ public void run() {
                                                         + authenticationRequest.getLogInfo());
 
                                         if (!StringExtensions.IsNullOrBlank(result.getAccessToken())) {
-                                            setItemToCache(authenticationRequest, result, true);
+                                            // TODO: Should pull out the interactive logic into another handler.
+                                            // https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/626
+                                            if (mTokenCacheAccessor != null) {
+                                                // Developer may pass null for the acquireToken flow.
+                                                mTokenCacheAccessor.updateTokenCache(authenticationRequest.getResource(), 
```

Comment:
If no userId in response, it means no idtoken returned. If there is no idtoken returned, how can we associate a token with a user? If this is the case, we'll just store the token with null user.